### DISCUSSION TITLE
enable checks for integer overflow/underflow in debug mode

### DISCUSSION THREAD

**Discussion by cristian-ilies-vasile** [vlang/v#11657] [Ideas]
See https://github.com/vlang/v/issues/4014

**Comment by dumblob** [comment]
There are many algorithms (typically those oriented on performance or low-level coding) relying on overflow/underflow (e.g. embedded development, security algorithms, hash/... algorithms, numerical computation algorithms, etc.).

So I'm afraid checking overflow/underflow would require **dedicated syntax** for (dis)allowing such checks.

**Comment by spytheman** [comment]
If you do:
```v
$if !no_overflow_checks ? { 
  // do slower checks here
}
```
... then when you compile with `v -d no_overflow_checks `, the checks will be skipped at compile time.

Of course it can do the opposite too:
```v
$if overflow_checks ? {
  // do slower checks
}
```
... so you can then opt in for slower but potentially safer code by passing `-d overflow_checks` during compilation.

I am not sure what approach is best for these.

**Comment by cristian-ilies-vasile** [comment]
My school of thinking was to go down the same route, more or less as for array's index checks.

I mean, let's the compiler add or not based on a command line flag (--enforce_overflow_check) if this type of verification will be injected in C code at compile time.
If the coder use V to animate sprites on a virtual world, then maybe is no need for such strict type of safety.

However for real life applications the developer will engage --enforce_overflow_check and automatically C code will be added to C source code by the V compiler.
